src/_pytest/fixtures.py

```diff
@@ -1037,7 +1037,7 @@ def execute(self, request: SubRequest) -> FixtureValue:
         return result
 
     def cache_key(self, request: SubRequest) -> object:
-        return request.param_index if not hasattr(request, "param") else request.param
+        return request.param_key
```

Comment:
(I need to remind myself a bit how CallSpec2 works with indirect parametrization...)